copylot/hardware/orca_camera/camera.py

```diff
@@ -77,3 +125,337 @@ def run(self, nb_frame: int = 100000):
+                                break
+
+                        dcam.cap_stop()
+
+                    dcam.buf_release()  # release buffer
+                else:
+                    print('-NG: Dcam.buf_alloc(3) fails with error {}'.format(dcam.lasterr()))
+                dcam.dev_close()
+
+        Dcamapi.uninit()
+
+    def live_capturing_return_images_get_ready(self, nb_buffer_frames=3, timeout_milisec=100):
+        """
+        this function only initialize the Dcamapi, allocate buffer, let capturing start for live capturing of images
+        using the orca camera. I'm calling it get_ready.
+
+        I'm now implementing some methods in a way that doesn't change anything in __init__() and run() that was already
+        there when I started.
+        Will re-organize these after discussion with AhmetCan.
+        -- Xiyu, 2022-11-16
+        :return:
+        """
+
+        if Dcamapi.init():
+            self.dcam = Dcam(self._camera_index)
+            if self.dcam.dev_open():
+                self.dcam.buf_alloc(nb_buffer_frames)
+                if self.dcam.cap_start():
+                    self.timeout_milisec = timeout_milisec
+                    self.dcam_status='started'
+
+    def live_capturing_return_images_capture_image(self):
+        """
+        this function captures an image and return the captured image as an ndarray.
+
+        I'm now implementing some methods in a way that doesn't change anything in __init__() and run() that was already
+        there when I started.
+        Will re-organize these after discussion with AhmetCan.
+        -- Xiyu, 2022-11-16
+
+        highly possible we will delete this method after the discussion.
+        :return:
+        """
+
+        if self.dcam_status=='started':
+            if self.dcam.wait_capevent_frameready(self.timeout_milisec) is not False:
+                print('capture the image')
+                self.data = self.dcam.buf_getlastframedata()
+            else:
+                dcamerr = self.dcam.lasterr()
+                if dcamerr.is_timeout():
+                    print('===: timeout')
+                else:
+                    print('Dcam.wait_event() fails with error{}'.format(dcamerr))
+        return self.data
+
+    def live_capturing_return_images_capture_end(self):
+        """
+        this function closes up the camera, and the buffer, and uninit the api.
+
+        I'm now implementing some methods in a way that doesn't change anything in __init__() and run() that was already
+        there when I started.
+        Will re-organize these after discussion with AhmetCan.
+        -- Xiyu, 2022-11-16
+
+        highly possible we will delete this method after the discussion.
+       :return:
+        """
+
+        self.dcam.cap_stop()
+        self.dcam.buf_release()
+        Dcamapi.uninit()
+
+    def set_configurations(self, camera_configs, camera_ids=[0]):
```

Comment:
I think we can refactor this method into something much more readable, let's chat on this in person some time @xiyuyi .